pool/pool.go

```diff
@@ -64,11 +71,13 @@ func getIndex(a []*PooledTransaction, target time.Time, low, high int) int {
 // Set inserts a transaction into the pool, returning
 // true if the Transaction was inserted (was valid).
 func (p *Pool) Set(t *blockchain.Transaction, bc *blockchain.BlockChain) bool {
-	if ok, _ := bc.ValidTransaction(t); ok {
+	if ok, err := bc.ValidTransaction(t); ok {
 		p.set(t)
```

Comment:
Will this succeed (ie add the transaction to the pool) if the transaction is valid wrt the blockchain but is not valid wrt some transaction already in the pool? ie. the input already exists in the pool. Does `p.set` handle that?